lib/THC/THCTensorIndex.cu

```diff
@@ -94,6 +95,124 @@ __global__ void indexCopyLargeIndex(TensorInfo<T, IndexType> dst,
   }
 }
 
+template <typename T, size_t n>
+struct AtomicAddIntegerImpl;
+
+template<typename T>
+struct AtomicAddIntegerImpl<T, 1> {
+  __device__ void operator()(T *address, T val) {
+    unsigned int * address_as_ui =
+        (unsigned int *) (address - ((size_t)address & 3));
+    unsigned int old = *address_as_ui, sum, newval, assumed;
```

Comment:
Please don't use comma lists for declaration, it is prone to confusion generally speaking. Especially here, since there is an initialization only for one variable along with declaration without initialization for others.

```
unsigned int old = *address_as_ui;
unsigned int sum;
unsigned int newval;
unsigned int assumed;
```